compiler/rustc_middle/src/ty/relate.rs

```diff
@@ -639,6 +639,15 @@ fn check_const_value_eq<R: TypeRelation<'tcx>>(
             get_slice_bytes(&tcx, a_val) == get_slice_bytes(&tcx, b_val)
         }
 
+        (ConstValue::ByRef { alloc: alloc_a, .. }, ConstValue::ByRef { alloc: alloc_b, .. })
+            if a.ty.is_ref() || b.ty.is_ref() =>
+        {
+            if a.ty.is_ref() && b.ty.is_ref() {
+                alloc_a == alloc_b
```

Comment:
This is going to have weird effects, but not worse than the Scalar::Ptr arm above. We should be using `deref_const` in both cases. But there's no need to try to fix this, we can just wait until I finally get [my valtree PR](https://github.com/rust-lang/rust/pull/83234) over the finish line.